src/paging.rs

```diff
@@ -39,13 +42,31 @@ pub fn setup() {
     // Point L4 at L3
     l4[0].set_addr(phys_addr(l3), pt_flags);
 
-    // Point Cr3 at L4
-    let (cr3_frame, cr3_flags) = Cr3::read();
-    let l4_frame = PhysFrame::from_start_address(phys_addr(l4)).unwrap();
-    if cr3_frame != l4_frame {
-        unsafe { Cr3::write(l4_frame, cr3_flags) };
+    if unsafe { core::arch::x86_64::__cpuid(7).ecx } & (1 << 16) != 0 {
+        // Point L5 at L4
+        l5[0].set_addr(phys_addr(l4), pt_flags);
+        // Point Cr3 at L5
+        let (cr3_frame, cr3_flags) = Cr3::read();
+        let l5_frame = PhysFrame::from_start_address(phys_addr(l5)).unwrap();
+        if cr3_frame != l5_frame {
+            unsafe { Cr3::write(l5_frame, cr3_flags) };
+        }
+        let mut cr4 = Cr4::read();
+        cr4.insert(Cr4Flags::L5_PAGING);
+        unsafe {
+            Cr4::write(cr4);
+        }
```

Comment:
Nit: Format like: `unsafe { Cr4::write(cr4) };` for consistency with the `Cr3` code.